docs/src/modules/components/DemoSandboxed.js

```diff
@@ -18,59 +17,47 @@ const styles = (theme) => ({
   },
 });
 
-function DemoFrame(props) {
-  const { children, classes, ...other } = props;
+function FramedDemo(props) {
+  const { children } = props;
+
+  const { document } = React.useContext(FrameContext);
+
   const theme = useTheme();
-  const [state, setState] = React.useState({
-    ready: false,
-  });
-  const instanceRef = React.useRef();
-
-  const handleRef = React.useCallback((ref) => {
-    instanceRef.current = {
-      contentDocument: ref ? ref.node.contentDocument : null,
-      contentWindow: ref ? ref.node.contentWindow : null,
-    };
-  }, []);
+  React.useEffect(() => {
+    document.body.dir = theme.direction;
+  }, [document, theme.direction]);
 
-  const onContentDidMount = () => {
-    setState({
-      ready: true,
+  const { jss, sheetsManager } = React.useMemo(() => {
+    return {
       jss: create({
         plugins: [...jssPreset().plugins, rtl()],
-        insertionPoint: instanceRef.current.contentWindow['demo-frame-jss'],
+        insertionPoint: document.head,
       }),
       sheetsManager: new Map(),
-      container: instanceRef.current.contentDocument.body,
-      window: () => instanceRef.current.contentWindow,
-    });
-  };
+    };
+  }, [document]);
+
+  const getWindow = React.useCallback(() => document.defaultView, [document]);
 
-  const onContentDidUpdate = () => {
-    instanceRef.current.contentDocument.body.dir = theme.direction;
-  };
+  return (
+    <StylesProvider jss={jss} sheetsManager={sheetsManager}>
+      {React.cloneElement(children, {
+        window: getWindow,
+      })}
+    </StylesProvider>
+  );
+}
+FramedDemo.propTypes = {
+  children: PropTypes.node,
+};
+
+function DemoFrame(props) {
+  const { children, classes, ...other } = props;
 
-  // NoSsr fixes a strange concurrency issue with iframe and quick React mount/unmount
```

Comment:
@oliviertassinari Do you recall the symptoms of that issue?

Reply:
I could find #13705

Reply:
We would need to make sure https://deploy-preview-20664--material-ui.netlify.app/zh/components/drawers/ displays without a crash.

Reply:
I'll check locally if `/zh/components/drawers/` crashes. That was the issue in #13705, right?

Reply:
Yes

Reply:
Does not crash locally in small and large viewporet.

I got warnings in concurrent mode for "state transition during render" which is indicative of some issue with react-frame-component. Might just be an issue with class components and concurrent mode. 

Hopefully this fixes concurrent mode issues as well. If not then at least let it not be a bug in react :)